sdk/package/lib/mainFn/Daemons.ts

```diff
@@ -413,26 +413,50 @@ export class Daemons<Manifest extends T.SDKManifest, Ids extends string>
 
   async term() {
     try {
-      for (let result of await Promise.allSettled(
-        this.healthDaemons.map((x) => x.term()),
-      )) {
-        if (result.status === "rejected") {
-          console.error(result.reason)
+      const remaining = new Set(this.healthDaemons)
```

Comment:
this is basically a topological sort on the dependencies, but in reverse.

daemons without dependants are terminated concurrently.